Test Failure: AutonomicalPolling1ServerTest.testAdd10RemoveAllImmediatelyAdd10PersistentExecs FFDC for PersistenceException

Test Failure: com.ibm.ws.concurrent.persistent.fat.autonomicalpolling1serv.AutonomicalPolling1ServerTest.testAdd10RemoveAllImmediatelyAdd10PersistentExecs

```
testAdd10RemoveAllImmediatelyAdd10PersistentExecs:junit.framework.AssertionFailedError: 2022-03-21-18:36:07:118 [Unexpected FFDC reporting javax.persistence.PersistenceException was found (count = 1): C:\Users\jazz_build\Build\jazz-build-engines\wasrtc\EBCPROD\build\dev\image\output\wlp\usr\servers\com.ibm.ws.concurrent.persistent.fat.autonomicalpolling1serv\logs\ffdc\ffdc_22.03.21_18.35.53.2.log
>------Start of DE processing------ = [3/21/22, 18:35:53:884 PDT]
>Exception = javax.persistence.PersistenceException
>Source = com.ibm.ws.concurrent.persistent.internal.InvokerTask
>probeid = 613
>Stack Dump = javax.persistence.PersistenceException: 
>Exception Description: Error binding to externally managed transaction
>Internal Exception: javax.transaction.RollbackException: Transaction rolled back
> at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:145)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
> at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:104)
> at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.persist(DatabaseTaskStore.java:1199)
> at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:577)
> at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:245)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> at java.base/java.lang.Thread.run(Thread.java:866)
>Caused by: javax.transaction.RollbackException: Transaction rolled back
> at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2231)
> at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2212)
> at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
, Unexpected FFDC reporting javax.transaction.RollbackException was found (count = 1): C:\Users\jazz_build\Build\jazz-build-engines\wasrtc\EBCPROD\build\dev\image\output\wlp\usr\servers\com.ibm.ws.concurrent.persistent.fat.autonomicalpolling1serv\logs\ffdc\ffdc_22.03.21_18.35.53.0.log
>------Start of DE processing------ = [3/21/22, 18:35:53:540 PDT]
>Exception = javax.transaction.RollbackException
>Source = com.ibm.tx.jta.TransactionImpl.registerSynchronization
>probeid = 2289
>Stack Dump = javax.transaction.RollbackException: Transaction rolled back
> at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2231)
> at com.ibm.tx.jta.impl.TransactionImpl.registerSynchronization(TransactionImpl.java:2212)
> at org.eclipse.persistence.internal.jpa.transaction.JTATransactionWrapper.registerIfRequired(JTATransactionWrapper.java:136)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.setJTATransactionWrapper(EntityManagerImpl.java:2168)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.detectTransactionWrapper(EntityManagerImpl.java:893)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.initialize(EntityManagerImpl.java:416)
> at org.eclipse.persistence.internal.jpa.EntityManagerImpl.<init>(EntityManagerImpl.java:409)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryDelegate.createEntityManagerImpl(EntityManagerFactoryDelegate.java:324)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManagerImpl(EntityManagerFactoryImpl.java:337)
> at org.eclipse.persistence.internal.jpa.EntityManagerFactoryImpl.createEntityManager(EntityManagerFactoryImpl.java:303)
> at com.ibm.wsspi.persistence.internal.PersistenceServiceUnitImpl.createEntityManager(PersistenceServiceUnitImpl.java:104)
> at com.ibm.ws.concurrent.persistent.db.DatabaseTaskStore.persist(DatabaseTaskStore.java:1199)
> at com.ibm.ws.concurrent.persistent.internal.InvokerTask.run(InvokerTask.java:577)
> at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:245)
> at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> at java.base/java.lang.Thread.run(Thread.java:866)
]
at componenttest.custom.junit.runner.FATRunner.blowup(FATRunner.java:302)
at componenttest.custom.junit.runner.FATRunner.access$600(FATRunner.java:67)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:252)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:320)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:173)
```

Failover tests run cleanly apart from the FFDC and warning in the logs, which are the sort of thing that it's normal to see if you are forcing failures that would cause a failover.

--- a/dev/com.ibm.ws.concurrent.persistent_fat_auto_pollinterval/fat/src/com/ibm/ws/concurrent/persistent/fat/autonomicalpolling1serv/AutonomicalPolling1ServerTest.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat_auto_pollinterval/fat/src/com/ibm/ws/concurrent/persistent/fat/autonomicalpolling1serv/AutonomicalPolling1ServerTest.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020 IBM Corporation and others.
+ * Copyright (c) 2020,2022 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -67,6 +67,7 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     public static void tearDown() throws Exception {
         try {
             server.stopServer(
+                    "CWWKC1503W", // task attempt rolled back due to PersistenceException
                     "CWWKC1505E", // task timed out, but will be retried
                     "J2CA0027E", "J2CA0079E", "J2CA0088W", "DSRA0304E", "DSRA0302E");
         } finally {
@@ -82,6 +83,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -156,6 +159,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -232,6 +237,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -319,6 +326,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -406,6 +415,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -483,6 +494,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -574,6 +587,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -636,6 +651,8 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // This later results in a XA_RBTIMEOUT (106) error on the transaction (javax.transaction.xa.XAException).
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
+        "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "javax.persistence.PersistenceException", // errors reported through JPA
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })

--- a/dev/com.ibm.ws.concurrent.persistent_fat_auto_pollinterval/fat/src/com/ibm/ws/concurrent/persistent/fat/autonomicalpolling1serv/AutonomicalPolling1ServerTest.java
+++ b/dev/com.ibm.ws.concurrent.persistent_fat_auto_pollinterval/fat/src/com/ibm/ws/concurrent/persistent/fat/autonomicalpolling1serv/AutonomicalPolling1ServerTest.java
@@ -84,7 +84,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -160,7 +162,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -238,7 +242,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -327,7 +333,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -416,7 +424,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -495,7 +505,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -588,7 +600,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })
@@ -652,7 +666,9 @@ public class AutonomicalPolling1ServerTest extends FATServletClient {
     // We recover/retry just fine, but this is a reason why Derby is not used for production.
     @AllowedFFDC({
         "jakarta.persistence.PersistenceException", // errors reported through Jakarta Persistence
+        "jakarta.transaction.RollbackException",
         "javax.persistence.PersistenceException", // errors reported through JPA
+        "javax.transaction.RollbackException",
         "javax.transaction.xa.XAException",
         "java.sql.SQLSyntaxErrorException" // if EclipseLink creates duplicate tables/constraints concurrently
     })

